modules/core/src/shaderlib/project/project-functions.js

```diff
@@ -116,31 +106,26 @@ export function projectPosition(position, params) {
     fromCoordinateOrigin
   } = normalizeParameters(params);
 
-  switch (coordinateSystem) {
-    case COORDINATE_SYSTEM.LNGLAT_OFFSETS:
-    case COORDINATE_SYSTEM.METER_OFFSETS: {
-      const worldPosition = getWorldPosition(position, {
-        viewport,
-        modelMatrix,
-        coordinateSystem: fromCoordinateSystem,
-        coordinateOrigin: fromCoordinateOrigin,
-        offsetMode: true
-      });
-      const originWorld = lngLatZToWorldPosition(coordinateOrigin, viewport, true);
-      vec3.sub(worldPosition, worldPosition, originWorld);
-
-      return worldPosition;
-    }
+  const {geospatialOrigin, shaderCoordinateOrigin, offsetMode} = getOffsetOrigin(
+    viewport,
+    coordinateSystem,
+    coordinateOrigin
+  );
 
-    case COORDINATE_SYSTEM.LNGLAT:
-    case COORDINATE_SYSTEM.CARTESIAN:
-    default:
-      return getWorldPosition(position, {
-        viewport,
-        modelMatrix,
-        coordinateSystem: fromCoordinateSystem,
-        coordinateOrigin: fromCoordinateOrigin,
-        offsetMode: false
-      });
+  const worldPosition = getWorldPosition(position, {
+    viewport,
+    modelMatrix,
+    coordinateSystem: fromCoordinateSystem,
+    coordinateOrigin: fromCoordinateOrigin,
+    offsetMode
+  });
+
+  if (offsetMode) {
+    const positionCommonSpace = viewport.projectPosition(
+      geospatialOrigin || shaderCoordinateOrigin
```

Comment:
Might be missing a usage, but could this be simplified by having the value of `geospatialOrigin` default to `shaderCoordinateOrigin` instead of `null`? And then having `getOffsetOrigin` just return one `offsetOrigin` value.

Reply:
`geospatialOrigin` is always in lnglat and used later to calculate units per pixel. I'd leave it as is for now for readability.